src/app/app.module.ts

```diff
@@ -22,16 +21,20 @@ import { TestExecutionServiceConfig } from '../service/execution/test.execution.
 import { IndexService } from 'service/index/index.service';
 import { XtextIndexService } from '../service/index/xtext.index.service';
 import { XtextIndexServiceConfig } from 'service/index/xtext.index.service.config';
+import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
+
+import { AppTokenStorage } from './app.token.storage';
+import { AuthInterceptor } from './auth.interceptor';
 
 const appRoutes: Routes = [
-    { path: '', component: AppComponent }
+  { path: '', component: AppComponent }
 ];
 
-export function authHttpServiceFactory(http: Http, options: RequestOptions) {
-  return new AuthHttp(new AuthConfig({
-    tokenName: 'token',
-    tokenGetter: (() => sessionStorage.getItem('token'))
-  }), http, options);
+export function loadConfig(oidcConfigService: OidcConfigService) {
+  console.log('APP_INITIALIZER STARTING');
+  return () => {
+    oidcConfigService.load_using_stsServer('http://accounts.google.com');
```

Comment:
You are right, the whole configuration of the openid provider should go into a configuration file. We should however do this as soon as the si internal provider is available and tested so that this configuration will hold all information to configure si or alternatively google. Right now the whole info is within this file (I colocated the information now, to have the information more closely together).